compiler/rustc_codegen_gcc/src/consts.rs

```diff
@@ -326,10 +326,10 @@ pub fn const_alloc_to_gcc<'gcc, 'tcx>(
                 interpret::Pointer::new(prov, Size::from_bytes(ptr_offset)),
                 &cx.tcx,
             ),
-            abi::Scalar::Initialized {
+            cx.tcx.mk_layout(LayoutS::scalar(cx, abi::Scalar::Initialized {
                 value: Primitive::Pointer(address_space),
                 valid_range: WrappingRange::full(dl.pointer_size),
-            },
+            }, ReprOptions::default())),
```

Comment:
... and here as well. It should be whatever the proper repr is, but I haven't quite figured out how to get it in this context.